fp-proj/src/app/core/services/authorization.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { OAuthService, UserInfo, OAuthEvent, EventType } from 'angular-oauth2-oidc';
+import { HttpClient } from '@angular/common/http';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+import {User} from '../../models/User'
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthorizationService {
+
+  private user = new BehaviorSubject<User>(null);
+  user$ = this.user.asObservable();
+
+  loggedIn: boolean
+  constructor( private oauthService: OAuthService,
+    private httpClient: HttpClient) {
+      this.oauthService.events.subscribe(({ type }: OAuthEvent) => {
+              this.FetchUser(type)
+      }
+    );
+  }
+
+  private FetchUser(type: EventType){
+    switch (type) {
+      case 'token_received':
+        {
+          // var claims = this.oauthService.getIdentityClaims();
+          // this.user.next(<User> {name: claims.nickname})
+
+           this.oauthService.loadUserProfile().then(u => {
+             this.user.next(<User> {name: u.nickname})})
+          break;
+
+
+        }
+      case 'logout':
+        {
+          this.user.next(null)
+          break;
+        }
+      }
+}
+
+  Login()
+  {
+   this.oauthService.initImplicitFlow();
+  }
+
+  Logout()
+  {
+    this.oauthService.logOut();
```

Comment:
logOut skasuje ci tylko token ze storage. ja bym gdzies w jakims BehaviorSubject zupdejtowal rowniez stan (np usera na null). Mozna tez zrobic this.oauthService.revokeTokenAndLogout(); - wtedy do serwera autoryzacyjnego pojdzie info ze ma cie wylogowac ze wszytskich miejsc gdzie wykrozystywany jest token i go uniewaznic